test/physics/em/PhotoelectricInteractor.test.cc

```diff
@@ -0,0 +1,237 @@
+
+        // Check secondaries
+        ASSERT_GT(2, interaction.secondaries.size());
+        if (interaction.secondaries.size() == 1)
+        {
+            const auto& electron = interaction.secondaries.front();
+            EXPECT_TRUE(electron);
+            EXPECT_EQ(pointers_.electron_id, electron.def_id);
+            EXPECT_GT(this->particle_track().energy().value(),
+                      electron.energy.value());
+            EXPECT_LT(0, electron.energy.value());
+            EXPECT_SOFT_EQ(1.0, celeritas::norm(electron.direction));
+        }
+
+        // Check conservation between primary and secondaries
+        // TODO: atomic relaxation has not been implemented. If the binding
+        // energy of the sampled shell is larger than the incident photon
+        // energy, or there is an energy balance following atomic deexcitation,
+        // the energy will be deposited locally.
+        // this->check_conservation(interaction);
+    }
+
+  protected:
+    std::shared_ptr<LivermoreParams>           livermore_params_;
+    celeritas::PhotoelectricInteractorPointers pointers_;
+    celeritas::LivermoreParamsPointers         data_;
+};
+
+//---------------------------------------------------------------------------//
+// TESTS
+//---------------------------------------------------------------------------//
+
+TEST_F(PhotoelectricInteractorTest, basic)
+{
+    // Reserve 4 secondaries
+    this->resize_secondaries(4);
+
+    // Sampled element
+    ElementDefId el_id{0};
+
+    // Create the interactor
+    PhotoelectricInteractor interact(pointers_,
+                                     data_,
+                                     el_id,
+                                     this->particle_track(),
+                                     this->direction(),
+                                     this->secondary_allocator());
+    RandomEngine&           rng_engine = this->rng();
+
+    std::vector<double> energy_electron;
+    std::vector<double> costheta_electron;
+
+    // Produce four samples from the original incident energy/dir
+    for (int i : celeritas::range(4))
+    {
+        Interaction result = interact(rng_engine);
+        SCOPED_TRACE(result);
+        this->sanity_check(result);
+        EXPECT_EQ(result.secondaries.data(),
+                  this->secondary_allocator().get().data() + i);
+
+        // Add actual results to vector
+        energy_electron.push_back(result.secondaries.front().energy.value());
+        costheta_electron.push_back(celeritas::dot_product(
+            result.secondaries.front().direction, this->direction()));
+    }
+
+    EXPECT_EQ(4, this->secondary_allocator().get().size());
+
+    // Note: these are "gold" values based on the host RNG.
+    const double expected_energy_electron[]
+        = {9.9964167, 9.9964167, 9.9964167, 9.9964167};
+    const double expected_costheta_electron[]
+        = {0.9989507514765, 0.9309974252831, 0.9901365221334, 0.9971118554745};
```

Comment:
I think your setup here is for 10 MeV photons, which is way too high for photoelectric, so I'm not sure these gold numbers are particular meaningful. I think you want to be in the few-KeV range where photoelectric dominates. Could you change the setup to do `.01` MeV and see if there's a more interesting distribution in the output values?

Reply:
Oh you're right I never changed the default energy.. this should definitely be lower.